### ISSUE TITLE
Keep having problems when PVR information is loaded (database gets corrupted somehow)

### ISSUE THREAD

**Issue by M-Reimer** [FernetMenta/vdr-plugin-vnsiserver#106]
I'm finally out of ideas about what could be the reason for a problem, I have on a "Kodi based" HTPC system.

VDR runs in background as pure TV backend (I don't care about the VDR OSD at all). Kodi is the only frontend of this system and both are connected via VNSI.

From time to time (actually pretty often, about once a day) the PVR information is not loaded at all. Yesterday, I saved the log files in hope that someone can help me with this.

The big problem is, that Kodi doesn't know about timers if the PVR info loading fails. So it will shut down after the "inactivity timeout" even if a recording is running and will also not set the next recording startup time.
[kodi.log](https://github.com/FernetMenta/vdr-plugin-vnsiserver/files/2108774/kodi.log)
[vdr-syslog.log](https://github.com/FernetMenta/vdr-plugin-vnsiserver/files/2108775/vdr-syslog.log)

**Comment by M-Reimer** [comment]
It is possible to revive from this situation when deleting the PVR database from Kodi settings. Sometimes this has to be done several times and yesterday Kodi even crashed in this process. And it doesn't help long. Only about one day before the same problem occurs again.

**Comment by FernetMenta** [comment]
I don't think this issue is related to vnsi. Looks like a general kodi addon handling and pvr issue. Those requests fail instantly:

```
23:10:48.929 T:139690224850688   ERROR: PVR - GetChannelGroupMembers - addon ':127.0.0.1:34890' returned an error: server error
23:10:48.929 T:139690224850688   ERROR: PVR - GetChannelGroupMembers - cannot get group members from client '1151313531': server error
23:10:48.940 T:139690224850688   ERROR: PVR - GetChannelGroupMembers - addon ':127.0.0.1:34890' returned an error: server error
23:10:48.940 T:139690224850688   ERROR: PVR - GetChannelGroupMembers - cannot get group members from client '1151313531': server error
23:10:48.950 T:139690224850688   ERROR: PVR - GetChannelGroupMembers - addon ':127.0.0.1:34890' returned an error: server error
23:10:48.950 T:139690224850688   ERROR: PVR - GetChannelGroupMembers - cannot get group members from client '1151313531': server error
23:10:48.960 T:139690224850688   ERROR: PVR - GetChannelGroupMembers - addon ':127.0.0.1:34890' returned an error: server error
23:10:48.960 T:139690224850688   ERROR: PVR - GetChannelGroupMembers - cannot get group members from client '1151313531': server error
```

Seems vnsi isn't even called (no vnsi errors logged). v17 was not very stable in regard of asynchronous pvr connection handling. I suggest that you try Kodi v18

**Comment by M-Reimer** [comment]
So in other words: There is a good chance that even a switch to tvheadend will not fix my issues...

I'll have a look into building Kodi 18 Arch Linux packages and give this a try.
